### ISSUE TITLE
CI is running too slow

### ISSUE THREAD

**Issue by Milan-Shah** [aws-amplify/amplify-ui#927]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### Please describe your bug.

CI is running slower than expected! It's currently running at 35~40 min. This can be shortened by:

- [ ] caching build artifacts
- [ ] remove duplicate/repeated builds
- [ ] run `yarn environements pull` in parallel
- [ ] Running e2e and docs test in parallel

Other CI enhancement ideas include:

- [ ] cancel concurrent ci runs in pull request
- [ ] #880 
- [ ] Conditional tests
